src/System.Diagnostics.Tracing/src/System/Diagnostics/Tracing/EventCounter.cs

```diff
@@ -92,10 +108,8 @@ private void Enqueue(float value)
                 {
                     // It is possible that two threads both think the buffer is full, but only one get to actually flush it, the other
                     // will eventually enter this code path and potentially calling Flushing on a buffer that is not full, and that's okay too.
-                    lock (_bufferedValues)
-                    {
+                    lock (this) // Lock the counter
```

Comment:
We generally discourage locking on `this` for exposed public types.  There's nothing else we can lock on?